This [short tutorial](https://niwdoog.github.io/Genome_Assembly_Long_Reads_Hi-C/) covers some basic analyses aiming to perform a chromosome-level scaffolding and annotation of a diploid plant using primarily data produced by third generation sequencing.

Table of contents:

[I] Preface
In a nutshell
Learning outcomes
Keywords
DOI
[II] Software & sequencing data
Note for Windows users
Specs of the server
Conda and setting an environment
Packages installation
Additional software (not in Conda)
Downloading raw sequence data
[III] Genome assembly
[Step 1 | NECAT]
[Step 2 | NextDenovo]
[Step 3 | Remove contaminants]
[Step 4 | NextPolish]
[Step 5 | Scaffolding]
[IV] Genome annotation
[Step 1 | Repeat identification and masking]
RepeatModeler
RepeatMasker
[Step 2 | Structural annotation]
[Step 3 | Functional annotation]
[V] Quality assessment & visualisation
[Step 1 | Qualimap]
[Step 2 | QUAST]
[Step 3 | k-mer analysis]
[Step 4 | BUSCO]
[Step 5 | IGV]
[Step 6 | Synteny]
[Step 7 | Circos]
Acknowledgements

[![DOI](https://zenodo.org/badge/543131281.svg)](https://zenodo.org/badge/latestdoi/543131281)
![](https://visitor-badge.deta.dev/badge?page_id=https://niwdoog.github.io/Genome_Assembly_Long_Reads_Hi-C/)
