[This short tutorial](https://niwdoog.github.io/Genome_Assembly_Long_Reads_Hi-C/) covers some basic analyses aiming to perform a chromosome-level scaffolding and annotation of a diploid plant using primarily data produced by third generation sequencing.

Table of contents:

[I] Preface<br/>
In a nutshell<br/>
Learning outcomes<br/>
Keywords<br/>
DOI<br/>
[II] Software & sequencing data<br/>
Note for Windows users<br/>
Specs of the server<br/>
Conda and setting an environment<br/>
Packages installation<br/>
Additional software (not in Conda)<br/>
Downloading raw sequence data<br/>
[III] Genome assembly<br/>
[Step 1 | NECAT]<br/>
[Step 2 | NextDenovo]<br/>
[Step 3 | Remove contaminants]<br/>
[Step 4 | NextPolish]<br/>
[Step 5 | Scaffolding]<br/>
[IV] Genome annotation<br/>
[Step 1 | Repeat identification and masking]<br/>
RepeatModeler<br/>
RepeatMasker<br/>
[Step 2 | Structural annotation]<br/>
[Step 3 | Functional annotation]<br/>
[V] Quality assessment & visualisation<br/>
[Step 1 | Qualimap]<br/>
[Step 2 | QUAST]<br/>
[Step 3 | k-mer analysis]<br/>
[Step 4 | BUSCO]<br/>
[Step 5 | IGV]<br/>
[Step 6 | Synteny]<br/>
[Step 7 | Circos]<br/>
Acknowledgements<br/>

[![DOI](https://zenodo.org/badge/543131281.svg)](https://zenodo.org/badge/latestdoi/543131281)
![](https://visitor-badge.deta.dev/badge?page_id=https://niwdoog.github.io/Genome_Assembly_Long_Reads_Hi-C/)
